File: restaurant_api/models/Menu.go
package models

type Menu struct {
	Id          int64  `gorm:"primaryKey" json:"id"`
	MenuName    string `gorm:"type:varchar(300);not null;default:null" json:"menu_name"`
	Description string `gorm:"type:text" json:"description"`
	Price       int    `gorm:"type:int;not null;default:null" json:"price"`
}

// Return Custom Table Name
func (Menu) TableName() string {
	return "tblMenu"
}

File: restaurant_mobile_app/lib/models/menu.dart
class Menu {
  final int Id;
  final String MenuName;
  final String Description;
  final double Price;

  Menu({
    required this.Id,
    required this.MenuName,
    required this.Description,
    required this.Price,
  });

  factory Menu.fromJson(Map<String, dynamic> json) {
    return Menu(
      Id: json['id'],
      MenuName: json['menu_name'],
      Description: json['description'],
      Price: json['price'].toDouble(),
    );
  }
}
